packages/airnode-node/src/workers/local-gateways/validation.ts

```diff
@@ -246,19 +270,71 @@ export function verifySignOevDataRequest(
     };
   }
 
-  const beaconsConsistent = allBeaconsConsistent(validDecodedBeacons);
-  if (!beaconsConsistent) {
+  if (!allBeaconsConsistent(validDecodedBeacons)) {
     return {
       success: false,
       statusCode: 400,
       error: { message: 'Inconsistent beacon data' },
     };
   }
 
+  const beaconsWithTemplateId = signedData.map((beacon) => {
+    const templateId = getExpectedTemplateIdV1({
+      airnodeAddress: beacon.airnodeAddress,
+      endpointId: beacon.endpointId,
+      encodedParameters: beacon.encodedParameters,
+    });
+
+    return { ...beacon, templateId };
+  });
+  const beaconIds = beaconsWithTemplateId.map((beacon) => deriveBeaconId(beacon.airnodeAddress, beacon.templateId));
+  // We are computing both update value and data feed ID in Airnode to prevent spoofing the signature.
+  const dataFeedId = beaconIds.length === 1 ? beaconIds[0] : deriveBeaconSetId(beaconIds);
+  const timestamp = calculateUpdateTimestamp(map(validDecodedBeacons, 'signedData.timestamp'));
+  const updateValue = calculateMedian(map(validDecodedBeacons, 'decodedValue'));
+
+  const goDeriveOevUpdateHash = goSync(() => {
```

Comment:
Deriving the update hash can fail for invalid requests, so it belongs to the validation.

Reply:
What do you consider an invalid request? The validation should prevent the processing of invalid requests. But the update hash derivation should happen only once you have a valid request, not as part of the validation.

Reply:
For example, when there is incorrect `updateId` (it needs to be encodable to bytes32). The validation can check whether the request has a good enough shape, but it's incomplete unless it actually derives the hash.

Reply:
That's like saying that you can't be sure about any processing unless it actually happens, so all the processing should be done as part of the validation. I think there should be a line and in this case, update hash derivation should be part of the processing and not part of the validation. For me personally is confusing to find this functionality as part of the validation process.

Reply:
I don't think there is a line. When we receive a request, we validate it and return 400 in case there is validation error. If not, we process the request and any error inside processing causes us to returns 500.

Reply:
I've added a comment to make the intention clearer.